src/node_worker.cc

```diff
@@ -775,6 +784,13 @@ void Worker::Exit(ExitCode code,
                   const char* error_code,
                   const char* error_message) {
   Mutex::ScopedLock lock(mutex_);
+
+  // Remove reference to the port used for inter-thread communication
+  {
+    Mutex::ScopedLock message_ports_lock(messagePortsMutex);
+    message_ports.erase(thread_id_.id);
```

Comment:
In case there are other cases to exit a worker (e.g. unhandled exception, kill from parent,...) such an erase needs to be also added in that places.